src/script/descriptor.cpp

```diff
@@ -1189,23 +1189,30 @@ class MiniscriptDescriptor final : public DescriptorImpl
 {
 private:
     miniscript::NodeRef<uint32_t> m_node;
+    miniscript::MiniscriptContext script_ctx;
 
 protected:
     std::vector<CScript> MakeScripts(const std::vector<CPubKey>& keys, Span<const CScript> scripts,
                                      FlatSigningProvider& provider) const override
     {
-        for (const auto& key : keys) provider.pubkeys.emplace(key.GetID(), key);
-        return Vector(m_node->ToScript(ScriptMaker(keys, miniscript::MiniscriptContext::P2WSH)));
+        for (const auto& key : keys) {
+            if (miniscript::IsTapscript(script_ctx)) {
+                provider.pubkeys.emplace(Hash160(XOnlyPubKey{key}), key);
```

Comment:
In ffdca88818f77fda06c6331df5b4bb408ab1833b "descriptor: parse Miniscript expressions within Taproot descriptors"

I don't think this is correct. In most places, we don't lookup xonly pubkeys by their hash. We look them up by getting the hashes of both the even and odd versions of the pubkey. So this would result in pubkeys that cannot be found.

It may be necessary to add this to deal with `pkh()`s, but I think we also need to add the hashes of the even and odd too.

Reply:
Happy to also emplace the hashes of the even and odd serializations, but if i remember correctly it seemed unnecessary to me at the time to hash the public key thrice. What's a usecase for this?

Also, since an x-only serialization is always available but it's not the case for an even or odd serialization should we always look up Taproot key by x-only serialization hash instead of having 3 different identifiers?

Reply:
I didn't get any response to my questions from Andrew but figured he thought my approach was fine since he then ACK'd the PR.

Reply:
I convinced myself that this is fine. IIRC, these pubkeys are actually never used anywhere relevant (I think the only actual usage is in importing a descriptor to a legacy wallet). In signing, the keys will always be in the `SigningProvider` anyways since they have to be provided externally to the descriptor. Since there's no way for a private key of any format to produce an `XOnlyPubKey`, this case is already being handled elsewhere in signing logic.